File: Database/Assembly_SRPG/QuestCondParam.cs
namespace SRPG
{
    using System;

    public class QuestCondParam
    {
        public string iname;
        public int plvmax;
        public int plvmin;
        public int ulvmax;
        public int ulvmin;
        public int[] elem;
        public bool isElemLimit;
        public string[] job;
        public PartyCondType party_type;
        public string[] unit;
        public ESex sex;
        public int rmax;
        public int rmin;
        public int hmax;
        public int hmin;
        public int wmax;
        public int wmin;
        public int[] jobset;
        public string[] birth;

        public QuestCondParam()
        {
            base..ctor();
            return;
        }

        public bool Deserialize(JSON_QuestCondParam json)
        {
            int num;
            int num2;
            int num3;
            int num4;
            int num5;
            int num6;
            this.iname = json.iname;
            this.plvmax = json.plvmax;
            this.plvmin = json.plvmin;
            this.ulvmax = json.ulvmax;
            this.ulvmin = json.ulvmin;
            this.sex = json.sex;
            this.rmax = json.rmax;
            this.rmin = json.rmin;
            this.hmax = json.hmax;
            this.hmin = json.hmin;
            this.wmax = json.wmax;
            this.wmin = json.wmin;
            num = 0;
            this.elem = new int[Enum.GetValues(typeof(EElement)).Length];
            this.elem[0] = num6 = json.el_none;
            num += num6;
            this.elem[1] = num6 = json.el_fire;
            num += num6;
            this.elem[2] = num6 = json.el_watr;
            num += num6;
            this.elem[3] = num6 = json.el_wind;
            num += num6;
            this.elem[4] = num6 = json.el_thdr;
            num += num6;
            this.elem[5] = num6 = json.el_lit;
            num += num6;
            this.elem[6] = num6 = json.el_drk;
            num += num6;
            this.isElemLimit = num > 0;
            num2 = 0;
            this.jobset = new int[4];
            this.jobset[num2++] = json.jobset1;
            this.jobset[num2++] = json.jobset2;
            this.jobset[num2++] = json.jobset3;
            if (json.job == null)
            {
                goto Label_01E0;
            }
            this.job = new string[(int) json.job.Length];
            num3 = 0;
            goto Label_01D2;
        Label_01BE:
            this.job[num3] = json.job[num3];
            num3 += 1;
        Label_01D2:
            if (num3 < ((int) this.job.Length))
            {
                goto Label_01BE;
            }
        Label_01E0:
            if (json.unit == null)
            {
                goto Label_0227;
            }
            this.unit = new string[(int) json.unit.Length];
            num4 = 0;
            goto Label_0219;
        Label_0205:
            this.unit[num4] = json.unit[num4];
            num4 += 1;
        Label_0219:
            if (num4 < ((int) this.unit.Length))
            {
                goto Label_0205;
            }
        Label_0227:
            if (json.birth == null)
            {
                goto Label_0274;
            }
            this.birth = new string[(int) json.birth.Length];
            num5 = 0;
            goto Label_0265;
        Label_024D:
            this.birth[num5] = json.birth[num5];
            num5 += 1;
        Label_0265:
            if (num5 < ((int) this.birth.Length))
            {
                goto Label_024D;
            }
        Label_0274:
            this.party_type = (Enum.IsDefined(typeof(PartyCondType), (int) json.party_type) == null) ? 0 : json.party_type;
            return 1;
        }
    }
}

File: Database/genConvert/QuestCondParam.py
def QuestCondParam(json):
    this={}#QuestCondParamjson)
    if 'iname' in json:
        this['iname'] = json['iname']
    if 'plvmax' in json:
        this['plvmax'] = json['plvmax']
    if 'plvmin' in json:
        this['plvmin'] = json['plvmin']
    if 'ulvmax' in json:
        this['ulvmax'] = json['ulvmax']
    if 'ulvmin' in json:
        this['ulvmin'] = json['ulvmin']
    if 'sex' in json:
        this['sex'] = ENUM['ESex'][json['sex']]
    if 'rmax' in json:
        this['rmax'] = json['rmax']
    if 'rmin' in json:
        this['rmin'] = json['rmin']
    if 'hmax' in json:
        this['hmax'] = json['hmax']
    if 'hmin' in json:
        this['hmin'] = json['hmin']
    if 'wmax' in json:
        this['wmax'] = json['wmax']
    if 'wmin' in json:
        this['wmin'] = json['wmin']
    #intnum1=0
    #this.elem=newint[Enum.GetValues(typeof(EElement)).Length]
    if 'el_none' in json:
        this['isElemLimit'] = num1++(this.elem[1]=json.el_fire)+(this.elem[2]=json.el_watr)+(this.elem[3]=json.el_wind)+(this.elem[4]=json.el_thdr)+(this.elem[5]=json.el_lit)+(this.elem[6]=json.el_drk)>0
    #intnum2=0
    #this.jobset=newint[4]
    #intjobset1=this.jobset
    #intindex1=num2
    #intnum3=index1+1
    #intjobset1_1=json.jobset1
    #jobset1[index1]=jobset1_1
    #intjobset2=this.jobset
    #intindex2=num3
    #intnum4=index2+1
    #intjobset2_1=json.jobset2
    #jobset2[index2]=jobset2_1
    #intjobset3=this.jobset
    #intindex3=num4
    #intnum5=index3+1
    #intjobset3_1=json.jobset3
    #jobset3[index3]=jobset3_1
    #if(json.job!=null)
        if 'job' in json:
            this['job'] = newstring[json['job'].Length]
        #for(intindex4=0index4<this.job.Length++index4)
        if 'job' in json:
            this['job'][4] = json['job'][index4]
    #if(json.unit!=null)
        if 'unit' in json:
            this['unit'] = newstring[json['unit'].Length]
        #for(intindex4=0index4<this.unit.Length++index4)
        if 'unit' in json:
            this['unit'][4] = json['unit'][index4]
    #if(json.birth!=null)
        if 'birth' in json:
            this['birth'] = newstring[json['birth'].Length]
        #for(intindex4=0index4<this.birth.Length++index4)
        if 'birth' in json:
            this['birth'][4] = json['birth'][index4]
    if 'party_type' in json:
        this['party_type'] = !Enum.IsDefined,(object)json['party_type'])?PartyCondType.None:(PartyCondType)json['party_type']
    #returntrue
return this
